capywise/lib/views/dashboard/dashboard_screen.dart

```diff
@@ -0,0 +1,354 @@
+        oldEnd: oldEnd,
+        oldTitle: oldTitle,
+      ),
+    );
+    if (updated == null) return;
+
+    // Update the event
+    setState(() {
+      _events[index]["title"] = updated["title"];
+      _events[index]["start"] = updated["start"];
+      _events[index]["end"] = updated["end"];
+    });
+    _sortEvents(); // re-sort after editing
+  }
+
+  // Format start/end as "Tue, Jan 10 • 3:00 PM - 6:00 PM"
+  String _formatDateTimeRange(DateTime start, DateTime end) {
+    final dayFormat = DateFormat('E, MMM d').format(start);
+    final startTime = DateFormat('h:mm a').format(start);
+    final endTime = DateFormat('h:mm a').format(end);
+    return "$dayFormat • $startTime - $endTime";
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return BasePage(
+      title: 'Dashboard',
+
+      middleContent: ListView.builder(
+        padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 8),
+        itemCount: _events.length,
+        itemBuilder: (context, index) {
+          final e = _events[index];
+          final start = e["start"] as DateTime;
+          final end = e["end"] as DateTime;
+          final dateTimeLabel = _formatDateTimeRange(start, end);
+
+          return ExpandableCard(
+            dateTimeLabel: dateTimeLabel,
+            title: e["title"],
+            tag: e["tag"],
+            isChecked: e["isChecked"] ?? false,
+            backgroundColor: e["color"],
+            onEditPressed: () => _editCard(index),
+            onDeletePressed: () => _deleteCard(index),
+          );
+        },
+      ),
+
+      // RIGHT SIDEBAR is optional
+      rightSidebar: Container(
+        margin: const EdgeInsets.all(12),
+        decoration: BoxDecoration(
+          color: Colors.white,
+          borderRadius: BorderRadius.circular(8),
+          boxShadow: const [
+            BoxShadow(
+              color: Colors.black12,
+              blurRadius: 4,
+              spreadRadius: 2,
+            ),
+          ],
+        ),
+        child: const Center(
+          child: Text("Right sidebar content"),
+        ),
+      ),
+      showRightSidebar: true,
+    );
+  }
+}
+
+/// A dialog to edit the event's date/time (start/end) and title.
+class _EditDialog extends StatefulWidget {
+  final DateTime oldStart;
+  final DateTime oldEnd;
+  final String oldTitle;
+
+  const _EditDialog({
+    Key? key,
+    required this.oldStart,
+    required this.oldEnd,
+    required this.oldTitle,
+  }) : super(key: key);
+
+  @override
+  State<_EditDialog> createState() => _EditDialogState();
+}
+
+class _EditDialogState extends State<_EditDialog> {
+  late TextEditingController _titleController;
+  late DateTime _localStart;
+  late DateTime _localEnd;
+
+  @override
+  void initState() {
+    super.initState();
+    _titleController = TextEditingController(text: widget.oldTitle);
+    _localStart = widget.oldStart;
+    _localEnd = widget.oldEnd;
+  }
+
+  @override
+  void dispose() {
```

Comment:
```suggestion
  void dispose() 
  {
```
comments for this possibly?